rendezvous/README.md

```diff
@@ -186,7 +186,46 @@ register under a namespace of interest (eg: the relay namespace). This
 can be mitigated by requiring a Proof of Work scheme for client
 registrations.
 
-This is TBD before finalizing the spec.
+If a rendezvous node decides that Proof of Work is required to complete
+a registration, it will send a response with the status code `E_POW_REQUIRED`
+together with a `challenge`.
+
+A client is then expected to follow up with a `ProofOfWork` message, computing
+a hash by concatenating the following:
+
+- The UTF8 bytes of the string `libp2p-rendezvous-pow`
+- The contents of `challenge`
+- The `ns` field
+- The `signedPeerRecord` field
+- The `PeerId` of the rendezvous point we are registering with
+- The `nonce` field
+
+The resulting byte buffer is hashed using SHA256.
+
+We define "difficulty" as the number of zero bytes at the front of the
+hash assuming a big endian encoding. If a rendezvous point considers
+the difficulty too low, it will decline a `ProofOfWork` message with
+a `RegisterResponse` and the status code set to `E_POW_DIFFICULTY_TOO_LOW` together
+with a new `challenge`.
+
+Once the difficulty requirement is met, a rendezvous point will respond with a
+status code of `OK`.
+
+To vary the difficulty, a client can manipulate the `nonce` field.
```

Comment:
In the current proposal, the rendezvous point never specifies a difficulty. Maybe it should but that is another field with need to include the message :)

It might be worth it because then the client doesn't need to "trial and error" in getting the difficulty right.

On the other hand, it doesn't really reduce complexity because we still need to cover the error case where a client ignores that field and sends a hash that doesn't satisfy the difficulty. Hence, the only benefit would be in less round-trips and less hashing on the client which sounds like a good idea.

> The client would vary or manipulate the nonce to produce a hash that satisfies the difficulty?

Yep, that is exactly what the spec says, maybe in a sort of contrived way: The `nonce` is the only field that can be varied by the client to produce a different hash.